Docs edits and additions after doing an initial setup (#579)
nix-community/lanzaboote: docs/explanation/troubleshooting.md

It is the most likely issue that Lanzaboote could not verify a cryptographic hash.
To recover from this, disable Secure Boot in your firmware settings.
Please file a bug, if you hit this issue.

## Failing to execute systemd-cryptenroll for measured boot

> If luks has been set up with the graphical installer it might have used `LUKS1` instead of the expected `LUKS2`.

Reviewer: nit: Can you fix `luks` -> `LUKS` everywhere for consistency? This sentence could be:

> Your LUKS-encrypted filesystem might use still be in the old (version 1) format, which is not supported.

Reviewer: There are also more occurrences later. The same also applies to PCRs, TPM and other abbreviations. If you refer to commands or Nix module options please use `code formatting`.

Check your luks version with:

```
$ cryptsetup luksDump /dev/sdX
LUKS header information
Version:       	2
Epoch:         	12
...
```

In that case you can convert in place while the partition is not in use with `cryptsetup convert --type luks2 /dev/sdX`. If this is your root partition you will need to live boot to convert your luks partition.

> ## Stuck with a pcrs policy error when doing nixos-rebuild

Reviewer: ```suggestion
## PCR policy error when running `nixos-rebuild`
```

Btw, how does this error message look like? Can we include it here?

Author: I could not reproduce it and got some help again from @nikstur to find it: `Failed to submit PCR policy to TPM: Remote address changed`.

I will add that to the section.

> In some cases the nixos-rebuild seems like it is stuck with policy violations while testing to add more entries in `boot.lanzaboote.measuredBoot.pcrs`.

Reviewer: Maybe:

```suggestion
In some cases, `nixos-rebuild` fails due to PCR policy violations while trying to add more entries in `boot.lanzaboote.measuredBoot.pcrs`.
```

> Remove `/var/lib/systemd/pcrlock.json` to have a nixos-rebuild re-initialize the on your filesystem documented tpm state and location.

Reviewer: ```suggestion
Remove `/var/lib/systemd/pcrlock.json` so `nixos-rebuild` will re-initialize ...
```

The rest of the sentence also looks a bit b0rken.

## Updating luks partition header and reset tpm pin

> If you want to change the tpm pin add `--wipe-slot=tpm2` to the cryptenroll command used to setup measured boot and execute again.

Reviewer: The other sections describe some error to be resolved. Why would I want to do this?

Author: This is the second part from the PCR policy error section, I will merge them together.

It is the most likely issue that Lanzaboote could not verify a cryptographic hash.
To recover from this, disable Secure Boot in your firmware settings.
Please file a bug, if you hit this issue.

## Failing to execute `systemd-cryptenroll` for measured boot

Your LUKS-encrypted filesystem might still use the old (version 1) format, which is not supported.
Check your LUKS version with:

```
$ cryptsetup luksDump /dev/sdX
LUKS header information

> Version:       	2

Reviewer 2: This should be the output for luks1 if you continue the next sentence with "in this case"

Epoch:         	12
...
```

> In that case you can convert in place while the partition is not in use with `cryptsetup convert --type LUKS2 /dev/sdX`. If this is your root partition you will need to live boot to convert your LUKS partition.

Reviewer 2: This should say "Live boot from a USB stick"

## PCR policy error when running `nixos-rebuild`

In some cases, `nixos-rebuild` fails due to PCR policy violations while trying to add more entries in `boot.lanzaboote.measuredBoot.pcrs`.
This error message has been observed before:
```
Failed to submit PCR policy to TPM: Remote address changed
```

Remove `/var/lib/systemd/pcrlock.json` and run nixos-rebuild to re-initialize the `pcrlock.json`.

> When PCR lock crashed it will not have re-enrolled. Follow up manually by running cryptenroll with `--wipe-slot=tpm2`:

Reviewer 2: This should say something slightly different: if the pcrlock.json has been removed and re-created you need to re-enroll the policy in your luks volume
